src/include/daos_task.h

```diff
@@ -171,20 +174,34 @@ typedef struct {
 	int			force;
 } daos_pool_destroy_t;
 
-/** pool connect args */
+/** pool connect by UUID args */
 typedef struct {
 	/** UUID of the pool. */
-	uuid_t			uuid;
+	uuid_t			 uuid;
 	/** Process set name of the DAOS servers managing the pool. */
 	const char		*grp;
 	/** Connect mode represented by the DAOS_PC_ bits. */
-	unsigned int		flags;
+	unsigned int		 flags;
 	/** Returned open handle. */
 	daos_handle_t		*poh;
 	/** Optional, returned pool information. */
 	daos_pool_info_t	*info;
 } daos_pool_connect_t;
 
+/** pool connect by label args */
+typedef struct {
+	/** Label of the pool. */
+	const char		*label;
+	/** Process set name of the DAOS servers managing the pool. */
+	const char		*grp;
+	/** Connect mode represented by the DAOS_PC_ bits. */
+	unsigned int		 flags;
+	/** Returned open handle. */
+	daos_handle_t		*poh;
+	/** Optional, returned pool information. */
+	daos_pool_info_t	*info;
+} daos_pool_connect_lbl_t;
```

Comment:
Why can't we just extend the existing `daos_pool_connect_t` struct to include the label field?

Reply:
Some clients such as the HDF5 daos vol connector use the task API and extending the existing structure causes an ABI break. Changing existing API/ABI is trying to be avoided in this patch. In earlier edits of this patch I had some related problems in this part of the code that resulted in segmentation faults in vol-daos tests.

Reply:
technically we could just land this PR with the vol tests disabled, and as soon as the master build becomes available, we can trigger the HDF5 VOL builds so that future PRs won't fail.
(this would also allow to insert the new symbols appropriately in the task table and not just at the end)

Reply:
@mchaarawi what would be the implications of adding such an ABI break for daos packaging, and the consumer packages? Would it require new package versions to be built, along with bumping the API version to a new major version 2?

Reply:
yes indeed. would require in this case any middleware using the task API to be rebuilt.

Reply:
OK, trying to avoid that in this first increment, so propose to keep the patch as currently constructed. Subsequently it is planned to evaluate how tools/middlewares could take advantage of the new API (and, if we do decide to rearrange the task table items, to rebuild for a new ABI).

Reply:
my thought was more to remove this new:
daos_pool_connect_lbl_t
and just add an optional 
const char *label
to
daos_pool_connect_t

I do understand the issue with the HDF5 vol. does it still break if you add the label as the last element in the struct?

Reply:
fwiw im fine to land this as is now, and do such changes in a future PR. would just need to avoid releasing 2.0 before making that change if we want it.

Reply:
I'm trying the simplification to put const char *label at the end of daos_pool_connect_t in an experimental PR that will run just the HDF5 vol tests and daos_core_test pool tests. https://github.com/daos-stack/daos/pull/5748

Reply:
That experiment seemed to work (daos_vol tests passed). I propose let's land this PR in current form, and I will try to roll this task API simplification into the follow-up PR for this work, https://github.com/daos-stack/daos/pull/5727 . That PR also adds container open bylabel API (and while doing that work I can also verify the simplification applies to dc_cont_open_t and won't require a new dc_cont_open_lbl_t).